tests/test_secure_config.py

```
@@ -391,23 +391,26 @@ def test_get_or_create_encryption_key_replace_failure_best_effort_cleanup(
             patch("secure_config.os.replace", side_effect=OSError("replace failed")),
-            patch.object(secure_config.Path, "unlink", _tracking_unlink),
         ):
+            # Should raise OSError from the failed os.replace
             with pytest.raises(OSError, match="Failed to write encryption key"):
                 get_or_create_encryption_key()
 
+        # Verify cleanup was attempted (unlink called during error handling)
         assert unlink_called
         if temp_file.exists():
-            temp_file.unlink()
+            original_unlink()
```

Flake8 (7.3.0)

[error] undefined name 'original_unlink'

(F821)